File: benchmarks/CWE-022-PathTraversal/path-traversal.js
// Vulnerable: Path traversal via user-controlled file path
const express = require('express');
const fs = require('fs');
const path = require('path');

const app = express();
const UPLOADS_DIR = '/var/data/uploads';

app.get('/files/:filename', (req, res) => {
  // Vulnerable: user-controlled filename joined without validation
  const filePath = path.join(UPLOADS_DIR, req.params.filename);
  const content = fs.readFileSync(filePath, 'utf-8');
  res.send(content);
});

app.get('/download', (req, res) => {
  // Vulnerable: direct user input in file path
  const file = req.query.path;
  res.download(file);
});

app.get('/reports/:year/:month', (req, res) => {
  // Vulnerable: path traversal via URL params
  const reportPath = `/data/reports/${req.params.year}/${req.params.month}/report.pdf`;
  res.sendFile(reportPath);
});

File: benchmarks/CWE-022-PathTraversal/path-traversal.py
# Vulnerable: Path traversal via user input
import os
from flask import Flask, request, send_file

app = Flask(__name__)
UPLOADS_DIR = '/var/data/uploads'

@app.route('/files/<path:filename>')
def get_file(filename):
    # Vulnerable: no path validation
    filepath = os.path.join(UPLOADS_DIR, filename)
    with open(filepath, 'r') as f:
        return f.read()

@app.route('/download')
def download():
    # Vulnerable: direct user input in path
    filepath = request.args.get('path')
    return send_file(filepath)

@app.route('/logs')
def get_logs():
    # Vulnerable: user input in file path
    date = request.args.get('date', 'today')
    log_file = f"/var/log/app/{date}.log"
    with open(log_file) as f:
        return f.read()
